python/hail/methods/family_methods.py

```diff
@@ -55,3 +56,172 @@ def trio_matrix(dataset, pedigree, complete_trios=False):
+
+    The copy state of a locus with respect to a trio is defined as follows,
+    where PAR is the `pseudoautosomal region
+    <https://en.wikipedia.org/wiki/Pseudoautosomal_region>`__ (PAR) defined by
+    the reference genome of the locus.
+
+    - HemiX -- in non-PAR of X, male child
+    - HemiY -- in non-PAR of Y, male child
+    - Auto -- otherwise (in autosome or PAR, or female child)
+
+    Any refers to :math:`\{ HomRef, Het, HomVar, NoCall \}` and ! denotes
+    complement in this set.
+
+    +--------+------------+------------+----------+------------------+
+    |Code    | Dad        | Mom        |     Kid  |   Copy State     |
+    +========+============+============+==========+==================+
+    |    1   | HomVar     | HomVar     | Het      | Auto             |
+    +--------+------------+------------+----------+------------------+
+    |    2   | HomRef     | HomRef     | Het      | Auto             |
+    +--------+------------+------------+----------+------------------+
+    |    3   | HomRef     |  ! HomRef  |  HomVar  | Auto             |
+    +--------+------------+------------+----------+------------------+
+    |    4   |  ! HomRef  | HomRef     |  HomVar  | Auto             |
+    +--------+------------+------------+----------+------------------+
+    |    5   | HomRef     | HomRef     |  HomVar  | Auto             |
+    +--------+------------+------------+----------+------------------+
+    |    6   | HomVar     |  ! HomVar  |  HomRef  | Auto             |
+    +--------+------------+------------+----------+------------------+
+    |    7   |  ! HomVar  | HomVar     |  HomRef  | Auto             |
+    +--------+------------+------------+----------+------------------+
+    |    8   | HomVar     | HomVar     |  HomRef  | Auto             |
+    +--------+------------+------------+----------+------------------+
+    |    9   | Any        | HomVar     |  HomRef  | HemiX            |
+    +--------+------------+------------+----------+------------------+
+    |   10   | Any        | HomRef     |  HomVar  | HemiX            |
+    +--------+------------+------------+----------+------------------+
+    |   11   | HomVar     | Any        |  HomRef  | HemiY            |
+    +--------+------------+------------+----------+------------------+
+    |   12   | HomRef     | Any        |  HomVar  | HemiY            |
+    +--------+------------+------------+----------+------------------+
+
+    This method only considers children with two parents and a defined sex.
+
+    This method assumes all contigs apart from X and Y are fully autosomal;
+    mitochondria, decoys, etc. are not given special treatment.
+
+    Parameters
+    ----------
+    dataset : :class:`.MatrixTable`
+        Dataset.
+    pedigree : :class:`.Pedigree`
+        Sample pedigree.
+
+    Returns
+    -------
+    (:class:`.Table`, :class:`.Table`, :class:`.Table`, :class:`.Table`)
+        Four tables with Mendel error statistics.
```

Comment:
", as detailed in notes?" I like cross-referencing things, in general, because it's nicer when skimming docs.